test/sync_fixtures.hpp

```diff
@@ -838,24 +835,16 @@ class MultiClientServerFixture {
 
     void run_client(int i)
     {
-        auto do_run_client = [this, i] {
-            auto sim = m_simulated_client_error_rates[i];
-            if (sim.first != 0) {
-                using sf = _impl::SimulatedFailure;
-                sf::RandomPrimeGuard pg(sf::sync_client__read_head, sim.first, sim.second,
-                                        random_int<uint_fast64_t>()); // Seed from global generator
-                m_clients[i]->run();
-            }
-            else {
-                m_clients[i]->run();
-            }
-            m_clients[i]->stop();
-        };
-        unit_test::TestContext& test_context = m_test_context;
-        if (CHECK_NOTHROW(do_run_client()))
-            return;
-        stop();
-        m_server_loggers[i]->error("Exception was throw from client[%1]'s event loop", i + 1);
+        auto sim = m_simulated_client_error_rates[i];
```

Comment:
you could defer to creating the client here. start() would then actually start the clients and the servers. This would also fix the hack in `Sync_AsyncWaitCancellation` in think.

Reply:
That was a great idea, but unfortunately, the client needs to be available when calling `make_session` or `make_bound_session`, so it needs to be created before `start()` is called.

However, I should push posting the simulated failure closer to where the client is created so it is done first.